flopy/mf6/mfbase.py

```diff
@@ -416,14 +417,9 @@ def set_sim_path(self, path, internal_use=False):
             if self.simulation is not None:
                 self.simulation.set_sim_path(path)
                 return
-        # recalculate paths for everything
-        # resolve path type
-        path = self.string_to_file_path(path)
-        if os.path.isabs(path):
-            self._sim_path = path
-        else:
-            # assume path is relative to working directory
-            self._sim_path = os.path.join(os.getcwd(), path)
+        # ensure that _sim_path is absolute
+        path = Path(path)
+        self._sim_path = path.absolute()
```

Comment:
Maybe worth calling `path.expanduser().absolute()` to get tilde expansion too? Just an extra convenience. Could do this in a followup PR, supporting `PathLike` more broadly throughout flopy seems worthwhile